plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/rule/InMemoryRuleProcessingService.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm.rule;
+
+import org.opensearch.plugin.wlm.rule.attribute_extractor.AttributeExtractor;
+import org.opensearch.plugin.wlm.rule.storage.AttributeValueStore;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * This class is responsible for managing in-memory view of Rules and Find matching Rule for the request
+ */
+public class InMemoryRuleProcessingService {
+
+    /**
+     * Default constructor
+     */
+    public InMemoryRuleProcessingService() {}
```

Comment:
This logic looks pretty generic to me which can in-turn be used by other plugins as well? If yes, should we plan to move this inside a common lib folder within Opensearch?

Reply:
Moved it to libs